src/components/ImageView/index.js

```diff
@@ -23,291 +22,250 @@ const propTypes = {
-            const moveY = this.state.initialY - y;
-            this.scrollableRef.scrollLeft = this.state.initialScrollLeft + moveX;
-            this.scrollableRef.scrollTop = this.state.initialScrollTop + moveY;
+            const moveX = initialX - x;
+            const moveY = initialY - y;
+            scrollableRef.current.scrollLeft = initialScrollLeft + moveX;
+            scrollableRef.current.scrollTop = initialScrollTop + moveY;
         }
 
-        this.setState((prevState) => ({isDragging: prevState.isMouseDown}));
-    }
-
-    imageLoad({nativeEvent}) {
-        this.setImageRegion(nativeEvent.width, nativeEvent.height);
-        this.setState({isLoading: false});
-    }
+        setIsDragging(isMouseDown);
+    };
 
-    imageLoadingStart() {
-        if (this.state.isLoading) {
+    useEffect(() => {
+        if (!isZoomed || !zoomDelta || !scrollableRef.current) {
             return;
         }
-        this.setState({isLoading: true, zoomScale: 0, isZoomed: false});
-    }
+        scrollableRef.current.scrollLeft = zoomDelta.offsetX;
+        scrollableRef.current.scrollTop = zoomDelta.offsetY;
+    }, [zoomDelta, isZoomed]);
 
-    render() {
-        if (this.canUseTouchScreen) {
-            return (
-                <View
-                    style={[styles.imageViewContainer, styles.overflowHidden]}
-                    onLayout={this.onContainerLayoutChanged}
-                >
-                    <Image
-                        source={{uri: this.props.url}}
-                        isAuthTokenRequired={this.props.isAuthTokenRequired}
-                        // Hide image until finished loading to prevent showing preview with wrong dimensions.
-                        style={this.state.isLoading ? undefined : [styles.w100, styles.h100]}
-                        // When Image dimensions are lower than the container boundary(zoomscale <= 1), use `contain` to render the image with natural dimensions.
-                        // Both `center` and `contain` keeps the image centered on both x and y axis.
-                        resizeMode={this.state.zoomScale > 1 ? Image.resizeMode.center : Image.resizeMode.contain}
-                        onLoadStart={this.imageLoadingStart}
-                        onLoad={this.imageLoad}
-                    />
-                    {this.state.isLoading && <FullscreenLoadingIndicator style={[styles.opacity1, styles.bgTransparent]} />}
-                </View>
-            );
+    useEffect(() => {
+        if (canUseTouchScreen) {
+            return;
         }
+        document.addEventListener('mousemove', trackMovement);
+        document.addEventListener('mouseup', trackPointerPosition);
+
+        return () => {
+            document.removeEventListener('mousemove', trackMovement);
+            document.removeEventListener('mouseup', trackPointerPosition);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I can't figure out what's the reasoning for silencing this warning... Could you explain it to me?

Reply:
I had some bugs related to this change, after some reconsideration, I removed this comment and fixed all errors